[vulkan] Use 2D texture types for convolution weights and biases

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86972
* #86971

Differential Revision: [D40385500](https://our.internmc.facebook.com/intern/diff/D40385500/)

Commit:
[vulkan] Use 2D texture types for convolution weights and biases



Differential Revision: [D40385500](https://our.internmc.facebook.com/intern/diff/D40385500/)

[ghstack-poisoned]

salilsdesai reviewed (approved):
![image](https://user-images.githubusercontent.com/25376640/195901563-eb077b9c-a722-4c97-b4f0-f9dc8968662b.png)

Commit:
Update on "[vulkan] Use 2D texture types for convolution weights and biases"


Differential Revision: [D40385500](https://our.internmc.facebook.com/intern/diff/D40385500/)

[ghstack-poisoned]
